File: pset1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    float changeindol;
    int coins = 0;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int pennie = 1;
    //ask a user for a number of changed owed
    do
    {
        changeindol = get_float("Change owed: \n");
    }
    while (changeindol < 0);
    //convert amount of dollars owed to amount of cents and round
    int cents = round(changeindol * 100);
    //check if you can give quarter
    while ((cents - quarter) >= 0)
    {
        coins += 1;
        cents = cents - quarter;
    }
    //if you can't give quarter start giving dimes
    while ((cents - dime) >= 0)
    {
        coins += 1;
        cents = cents - dime;
    }
    //if you can't give quarter start giving nickels
    while ((cents - nickel) >= 0)
    {
        coins += 1;
        cents = cents - nickel;
    }
    //if you can't give quarter start giving pennies
    while ((cents - pennie) >= 0)
    {
        coins += 1;
        cents = cents - pennie;
    }

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float
# function that repeats asking for a POSITIVE float if it has not yet been provided


def get_positive_float():
    while True:
        f = get_float("Change owed: ")
        if f >= 0:
            break
    return f


# ask a user for a positive amount of change to be given in dollars
change = get_positive_float()
# convert the dollars to cents, and round to the nearest integer
cents = round(change * 100)
# initialize a coin counter
coins = 0
# start giving change with quartes, then proceed to dimes, nickels and pennies
while (cents - 25) >= 0:
    coins += 1
    cents -= 25
while (cents - 10) >= 0:
    coins += 1
    cents -= 10
while (cents - 5) >= 0:
    coins += 1
    cents -= 5
while (cents - 1) >= 0:
    coins += 1
    cents -= 1
print(coins)